lib/agent.js

```diff
@@ -261,6 +262,17 @@ class Agent {
             agentVersion: this.config.version,
             licensed: this.config.licensed
         }
+        if (this.launcher?.readPackage) {
+            const { modules } = this.launcher.readPackage()
+            if (module['node-red'] !== 'latest') {
```

Comment:
Since the goal is to get a semver, should we be testing `modules['node-red']` is a semver?

e.g. something like: `semver.valid(modules['node-red'])` instead?

Reply:
I was trying to not add any more dependencies

Reply:
semver is already a dependency: https://github.com/FlowFuse/device-agent/blob/de9bd72af2ba5ca8311adb9eb97ca15c4a4c4f70/package.json#L45